lms/validation/_helpers.py

```diff
@@ -0,0 +1,14 @@
+import inspect
+
+__all__ = ["instantiate_schema"]
+
+
+def instantiate_schema(schema, request):
+    # Schemas can just be dicts or they can be actual
+    # marshmallow.Schema subclasses, in which case we need to
+    # instantiate the class.
+    if inspect.isclass(schema):
+        schema = schema()
+        schema.context["request"] = request
```

Comment:
Field validator methods in schema classes will be able to access the request as `self.context["request"]`. This enables a validator method for the OAuth 2 state param to not only check that state is present and is a string, but that it matches the saved state in `request.session`.